core/primitives/src/epoch_manager.rs

```diff
@@ -156,19 +178,25 @@ impl AllEpochConfig {
 
     pub fn for_protocol_version(&self, protocol_version: ProtocolVersion) -> EpochConfig {
         if self.config_store.is_some() {
-            self.config_store.as_ref().unwrap().get_config(protocol_version).as_ref().clone()
+            let mut config =
+                self.config_store.as_ref().unwrap().get_config(protocol_version).as_ref().clone();
+            // TODO(#11265): epoch length is overridden in many tests so we
+            // need to support it here. Consider removing `epoch_length` from
+            // EpochConfig.
```

Comment:
I do not get this why epoch_length is special here. it is already overriden through test_epoch_config so it will be in the configs returned by the EpochConfigStore right?, so why do we need to keep a local field for epoch_length here?

Reply:
Many tests have setup pattern like `TestEnvBuilder::new().epoch_length(5)...`.
This invalidates epoch lengths given in default `EpochConfigStore`s.
So override must happen somewhere. We could put epoch length to `new` or remove out of `EpochConfig`, but that means more lines changes which I try to minimise.

Reply:
got it, we can address epoch length later, it is not a blocker for this PR from my side.